go/vt/discovery/endpointwatcher.go

```diff
@@ -0,0 +1,104 @@
+package discovery
+
+import (
+	"time"
+
+	log "github.com/golang/glog"
+	pbt "github.com/youtube/vitess/go/vt/proto/topodata"
+	"github.com/youtube/vitess/go/vt/topo"
+	"golang.org/x/net/context"
+)
+
+// NewEndPointWatcher returns an EndPointWatcher and starts refreshing.
+func NewEndPointWatcher(topoServer topo.Server, hc HealthCheck, cellsToWatch []string, refreshInterval time.Duration) *EndPointWatcher {
+	epw := &EndPointWatcher{
+		topoServer:      topoServer,
+		hc:              hc,
+		cellsToWatch:    cellsToWatch,
+		refreshInterval: refreshInterval,
+		endPoints:       make(map[string]*pbt.EndPoint),
```

Comment:
How about keeping a map per cell? I feel like they're separate data sets anyway.

Actually, better yet, have this guy just watch a single cell, and have vtgate take the list of cells to watch, and it can create multiple instances of this guy. I feel it would be clearer.